src/gpl/src/nesterovBase.h

```diff
@@ -1123,4 +1133,37 @@ class biNormalParameters
   float uy;
 };
 
+struct GCellIndexHandle
```

Comment:
Why is it a struct and not a class with private members?  

Do clients need access to the index?  If not I would name it GCellHandle.